src/materialized/src/version_check.rs

```diff
@@ -50,34 +53,66 @@ pub async fn check_version_loop(telemetry_url: String, cluster_id: String) {
         Ok(_) => (),
         Err(e) => debug!("unable to parse fetched latest version: {}", e),
     }
+
+    loop {
+        tokio_sleep(TokioDuration::from_secs(TELEMETRY_FREQUENCY_SECS)).await;
+
+        fetch_latest_version(&version_url, start_time).await;
```

Comment:
We need to make sure this doesn't spam the log with "a new version of materialized is available" messages. Currently, the `check_version_loop` only compares the version received against the version running. This logic needs to also check for whether or not a message has been printed for the new version.

Reply:
I don't think it will spam the log? The print is only done once above, and then the loop just calls the fetch function which doesn't do any logging.

Reply:
Oh, I see. I misunderstood when the new version was being printed. That raises a new question: we only wish to print the new version message if and only if there is a new version at `materialized` startup, correct? We don't wish to print a message if a new version is released some time after `materialized` starts up?

Reply:
:shrug: The initial work was spec'd out by @awang. Maybe he has thoughts.